src/internal/incompatibility.rs

```diff
@@ -43,28 +43,25 @@ enum Kind<P: Package, V: Version> {
     /// Initial incompatibility aiming at picking the root package for the first decision.
     NotRoot(P, V),
     /// There are no versions in the given range for this package.
-    NoVersions(P, Range<V>),
+    NoVersions(P),
```

Comment:
I think things like `NoVersion(P)` might imply that there is no version at all for package P while actually that's limited to a given range.